usim/py/_awaitable.py

```diff
@@ -0,0 +1,60 @@
+from typing import Awaitable, TypeVar, Generic
+from .. import Flag, until
+
+
+R = TypeVar('R')
+
+
+class AwaitableEvent(Generic[R]):
+    """
+    Wraps an Awaitable as a SimPy :py:class:`~usim.py.events.Event`
+
+    This type only exists for internal usage to allow a
+    :py:class:`~usim.py.events.Process` to ``yield`` a :py:class:`~.Notification`
+    or :term:`activity`.
+    It is not intended for manual instantiation and does not provide the ``Event``
+    interface beyond what is needed internally.
+
+    If you encounter a :py:class:`~.NotificationEvent` during a simulation,
+    **this is a bug**.
```

Comment:
We could leave a comment and link to the ticket system?

Reply:
Yes, that is true. Will do.

Reply:
Added link to GitHub issues.